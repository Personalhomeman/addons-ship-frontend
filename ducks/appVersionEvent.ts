import { createReducer } from 'deox';

import { AppVersionEvent } from '@/models';
import fetchAppVersionEvents from './appVersion/fetchAppVersionEvents';

const defaultState: AppVersionEventsState = [];

export type AppVersionEventsState = AppVersionEvent[];
export const appVersionEvents = createReducer(defaultState, handleAction => [
  handleAction(fetchAppVersionEvents.complete, (_, { payload }) => payload)
]);
