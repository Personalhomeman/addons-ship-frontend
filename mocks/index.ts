import { AppVersion, AppVersionEvent } from '@/models';
import { Settings } from '@/models/settings';
import { TestDevice } from '@/models/test-device';

export const mockAppVersions: AppVersion[] = [
  {
    id: 123,
    appSlug: 'app-slug-123',
    version: '1.0.3',
    platform: 'ios',
    buildNumber: 32,
    buildSlug: '13245ads',
    lastUpdate: new Date('2018-09-22T12:42:31Z'),
    description: 'Some semi-long description',
    whatsNew: 'Some longer whats-new',
    minimumOs: '10.2',
    minimumSdk: 'dunno',
    packageName: 'com.package.name',
    certificateExpires: new Date('2018-09-22T12:42:31Z'),
    bundleId: 'bundle-id',
    size: 4567034,
    supportedDeviceTypes: ['iphone', 'ipad'],
    distributionType: 'development',
    iconUrl: 'http://placekitten.com/160/160',
    appName: 'Standup Timer',
    promotionalText: 'Promotional Text',
    keywords: 'Keywords',
    reviewNotes: 'Review Notes',
    supportUrl: 'https://bitrise.io/support',
    marketingUrl: 'https://bitrise.io/marketing',
    scheme: 'piramid',
    configuration: 'canary',
    publicInstallPageURL: 'https://bitrise.io/app/8b334705d8e78276'
  },
  {
    id: 456,
    appSlug: 'app-slug-456',
    version: '1.0.3',
    platform: 'ios',
    buildNumber: 31,
    buildSlug: '13245ads',
    lastUpdate: new Date('2018-09-22T12:42:31Z'),
    description: 'Some semi-long description',
    whatsNew: 'Some longer whats-new',
    minimumOs: '10.2',
    minimumSdk: 'dunno',
    packageName: 'com.package.name',
    certificateExpires: new Date('2018-09-21T12:42:31Z'),
    bundleId: 'bundle-id',
    size: 4567034,
    supportedDeviceTypes: ['iphone', 'ipad'],
    distributionType: 'development',
    iconUrl: 'http://placekitten.com/160/160',
    appName: 'Standup Timer',
    promotionalText: 'Promotional Text',
    keywords: 'Keywords',
    reviewNotes: 'Review Notes',
    supportUrl: 'https://bitrise.io/support',
    marketingUrl: 'https://bitrise.io/marketing',
    scheme: 'piramid',
    configuration: 'canary',
    publicInstallPageURL: 'https://bitrise.io/app/8b334705d8e78276'
  },
  {
    id: 789,
    appSlug: 'app-slug-789',
    version: '1.0.3',
    platform: 'ios',
    buildNumber: 30,
    buildSlug: '13245ads',
    lastUpdate: new Date('2018-09-22T12:42:31Z'),
    description: 'Some semi-long description',
    whatsNew: 'Some longer whats-new',
    minimumOs: '10.2',
    minimumSdk: 'dunno',
    packageName: 'com.package.name',
    certificateExpires: new Date('2018-09-20T12:42:31Z'),
    bundleId: 'bundle-id',
    size: 4567034,
    supportedDeviceTypes: ['iphone', 'ipad'],
    distributionType: 'development',
    iconUrl: 'http://placekitten.com/160/160',
    appName: 'Standup Timer',
    promotionalText: 'Promotional Text',
    keywords: 'Keywords',
    reviewNotes: 'Review Notes',
    supportUrl: 'https://bitrise.io/support',
    marketingUrl: 'https://bitrise.io/marketing',
    scheme: 'piramid',
    configuration: 'canary',
    publicInstallPageURL: 'https://bitrise.io/app/8b334705d8e78276'
  },
  {
    id: 135,
    appSlug: 'app-slug-135',
    version: '1.0.2',
    platform: 'ios',
    buildNumber: 29,
    buildSlug: '13245ads',
    lastUpdate: new Date('2018-09-22T12:42:31Z'),
    description: 'Some semi-long description',
    whatsNew: 'Some longer whats-new',
    minimumOs: '10.2',
    minimumSdk: 'dunno',
    packageName: 'com.package.name',
    certificateExpires: new Date('2018-09-29T12:42:31Z'),
    bundleId: 'bundle-id',
    size: 4567034,
    supportedDeviceTypes: ['iphone', 'ipad'],
    distributionType: 'development',
    iconUrl: 'http://placekitten.com/160/160',
    appName: 'Standup Timer',
    promotionalText: 'Promotional Text',
    keywords: 'Keywords',
    reviewNotes: 'Review Notes',
    supportUrl: 'https://bitrise.io/support',
    marketingUrl: 'https://bitrise.io/marketing',
    scheme: 'piramid',
    configuration: 'canary',
    publicInstallPageURL: 'https://bitrise.io/app/8b334705d8e78276'
  },
  {
    id: 246,
    appSlug: 'app-slug-246',
    version: '1.0.2',
    platform: 'ios',
    buildNumber: 28,
    buildSlug: '13245ads',
    lastUpdate: new Date('2018-09-22T12:42:31Z'),
    description: 'Some semi-long description',
    whatsNew: 'Some longer whats-new',
    minimumOs: '10.2',
    minimumSdk: 'dunno',
    packageName: 'com.package.name',
    certificateExpires: new Date('2018-09-28:42:31Z'),
    bundleId: 'bundle-id',
    size: 4567034,
    supportedDeviceTypes: ['iphone', 'ipad'],
    distributionType: 'development',
    iconUrl: 'http://placekitten.com/160/160',
    appName: 'Standup Timer',
    promotionalText: 'Promotional Text',
    keywords: 'Keywords',
    reviewNotes: 'Review Notes',
    supportUrl: 'https://bitrise.io/support',
    marketingUrl: 'https://bitrise.io/marketing',
    scheme: 'piramid',
    configuration: 'canary',
    publicInstallPageURL: 'https://bitrise.io/app/8b334705d8e78276'
  }
];

export const mockAppVersion = mockAppVersions[0];
export const mockAppVersionWithoutPublicPage = {
  id: 123,
  appSlug: 'app-slug-123',
  version: '1.0.3',
  platform: 'ios',
  buildNumber: 32,
  buildSlug: '13245ads',
  lastUpdate: new Date('2018-09-22T12:42:31Z'),
  description: 'Some semi-long description',
  whatsNew: 'Some longer whats-new',
  minimumOs: '10.2',
  minimumSdk: 'dunno',
  packageName: 'com.package.name',
  certificateExpires: new Date('2018-09-22T12:42:31Z'),
  bundleId: 'bundle-id',
  size: 4567034,
  supportedDeviceTypes: ['iphone', 'ipad'],
  distributionType: 'development',
  iconUrl: 'http://placekitten.com/160/160',
  appName: 'Standup Timer',
  promotionalText: 'Promotional Text',
  keywords: 'Keywords',
  reviewNotes: 'Review Notes',
  supportUrl: 'https://bitrise.io/support',
  marketingUrl: 'https://bitrise.io/marketing',
  scheme: 'piramid',
  configuration: 'canary',
  publicInstallPageURL: undefined
};

export const mockAndroidAppVersion = {
  id: 123,
  appSlug: 'app-slug-123',
  version: '1.0.3',
  platform: 'android',
  buildNumber: 32,
  buildSlug: '13245ads',
  lastUpdate: new Date('2018-09-22T12:42:31Z'),
  shortDescription: 'Some short description',
  description: 'Some semi-long description',
  whatsNew: 'Some longer whats-new',
  versionCode: '165161631',
  minimumSdk: '26',
  packageName: 'com.package.name',
  certificateExpires: new Date('2018-09-22T12:42:31Z'),
  size: 4567034,
  supportedDeviceTypes: ['phone', 'tablet', 'android-tv', 'wear-os'],
  iconUrl: 'http://placekitten.com/160/160',
  appName: 'Standup Timer',
  supportUrl: 'https://bitrise.io/support',
  marketingUrl: 'https://bitrise.io/marketing',
  module: 'Celeste-config',
  variant: 'Fennek',
  buildType: 'Celeste-config',
  publicInstallPageURL: 'https://bitrise.io/app/8b334705d8e78276'
};

export const mockTestDevices: TestDevice[] = [
  { deviceId: 'some-device-id-01', deviceType: 'ios', owner: 'test-user-1' },
  { deviceId: 'some-device-id-02', deviceType: 'android', owner: 'test-user-1' },
  { deviceId: 'some-device-id-03', deviceType: 'ios', owner: 'test-user-2' }
];

const provProfiles = [{ name: 'abcdefghijkl1234567' }, { name: 'abcdefghijkl12345671566' }];
const certificates = [
  { name: 'iPhone Developer: John Doe (ABCD1234)' },
  { name: 'iPhone Developer: John Doe (ABCD5678)' }
];
const keystoreFiles = [{ name: 'abcdefghijkl1234567' }, { name: 'abcdefghijkl12345671566' }];
const serviceAccountJsonFiles = [{ name: 'abcdefghijkl1234567' }, { name: 'abcdefghijkl12345671566' }];

export const mockSettings: Settings = {
  projectType: 'other',
  iosSettings: {
    artifactExposingWorkflows: 'All',
    appleDeveloperAccountEmail: 'Fill',
    appSku: 'Fill',
    appSpecificPassword: 'Fill',
    selectedProvProfile: provProfiles[0],
    selectedCertificate: certificates[1]
  },
  androidSettings: {
    artifactExposingWorkflows: 'All',
    track: 'Release',
    selectedKeystoreFile: keystoreFiles[0],
    selectedServiceAccountJsonFile: serviceAccountJsonFiles[1]
  },
  provProfiles: provProfiles,
  certificates: certificates,
  keystoreFiles: keystoreFiles,
  serviceAccountJsonFiles: serviceAccountJsonFiles
};

export const mockFinishedAppVersionEvent: AppVersionEvent = {
  id: '1234',
  status: 'finished',
  text: 'Event finished successfully!',
  createdAt: new Date('2018-09-20T12:42:31Z'),
  updatedAt: new Date('2018-09-21T12:42:31Z'),
  logDownloadUrl: 'https://www.bitrise.io/assets/svg/logo-bitrise.svg'
};

export const mockInProgressAppVersionEvent: AppVersionEvent = {
  id: '1234',
  status: 'in-progress',
  text: 'Doing something...',
  createdAt: new Date('2018-09-22T12:42:31Z'),
  updatedAt: new Date('2018-09-25T12:42:31Z'),
  logDownloadUrl: 'https://www.bitrise.io/assets/svg/logo-bitrise.svg'
};

export const mockFailedAppVersionEvent: AppVersionEvent = {
  id: '1234',
  status: 'failed',
  text: 'An error occured.',
  createdAt: new Date('2018-09-26T12:42:31Z'),
  updatedAt: new Date('2018-09-27T12:42:31Z'),
  logDownloadUrl: 'https://www.bitrise.io/assets/svg/logo-bitrise.svg'
};

export const mockAppVersionEvents: AppVersionEvent[] = [
  mockFinishedAppVersionEvent,
  mockInProgressAppVersionEvent,
  mockFailedAppVersionEvent
];
