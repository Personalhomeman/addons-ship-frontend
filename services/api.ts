import fetch, { Request, RequestInit, Response } from 'node-fetch';
import { AppVersion } from '@/models';

export type Fetch = (url: string | Request, init?: RequestInit) => Promise<Response>;
export type APIConfig = {
  url: string;
};

class APIService {
  constructor(private fetch: Fetch, private config: APIConfig) {}

  async getAppVersion(appSlug: string, versionId: string): Promise<AppVersion> {
    const resp = await {
      data: {
        id: 123,
        version: '1.0.3',
        platform: 'ios',
        build_number: '32',
        build_slug: '13245ads',
        last_update: '2018-09-22T12:42:31Z',
        description: 'Some semi-long description',
        whats_new: 'Some longer whats-new',
        minimum_os: '10.2',
        minimum_sdk: 'dunno',
        package_name: 'com.package.name',
        certificate_expires: '2018-09-22T12:42:31Z',
        bundle_id: 'bundle-id',
        size: 4567034,
        supported_device_types: ['iphone', 'ipad'],
        distribution_type: 'development',
        icon_url: 'http://placekitten.com/160/160',
        app_name: 'Standup Timer',
        promotional_text: 'Promotional Text',
        keywords: 'Keywords',
        review_notes: 'Review Notes',
        support_url: 'https://bitrise.io/support',
        marketing_url: 'https://bitrise.io/marketing',
        scheme: 'piramid',
        configuration: 'canary'
      }
    };

    const { data } = resp;

    return {
      id: data.id,
      version: data.version,
      platform: data.platform,
      buildNumber: parseInt(data.build_number, 10),
      buildSlug: data.build_slug,
      lastUpdate: new Date(data.last_update),
      description: data.description,
      whatsNew: data.whats_new,
      minimumOs: data.minimum_os,
      minimumSdk: data.minimum_sdk,
      packageName: data.package_name,
      certificateExpires: new Date(data.certificate_expires),
      bundleId: data.bundle_id,
      size: data.size,
      supportedDeviceTypes: data.supported_device_types,
      distributionType: data.distribution_type,
      iconUrl: data.icon_url,
      appName: data.app_name,
      promotionalText: data.promotional_text,
      keywords: data.keywords,
      reviewNotes: data.review_notes,
      supportUrl: data.support_url,
      marketingUrl: data.marketing_url,
      scheme: data.scheme,
      configuration: data.configuration
    };
  }
}

export default new APIService(fetch, {
  url: 'todo: get from config'
});
