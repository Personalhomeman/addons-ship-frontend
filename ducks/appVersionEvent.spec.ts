jest.mock('@/services/ship-api');

import configureMockStore, { MockStoreCreator, MockStoreEnhanced } from 'redux-mock-store';
import thunk from 'redux-thunk';

import api from '@/services/ship-api';

import { appVersionEvents, fetchAppVersionEvents } from './appVersionEvent';
import { mockAppVersionEvents } from '@/mocks';

describe('appVersion', () => {
  let mockStore: MockStoreCreator;
  beforeEach(() => {
    mockStore = configureMockStore([thunk]);
  });

  describe('reducer', () => {
    it('loads app version events', () => {
      const state = appVersionEvents(undefined, fetchAppVersionEvents.complete(mockAppVersionEvents));

      expect(state).toMatchSnapshot();
    });
  });

  describe('fetchAppVersionEvents', () => {
    let store: MockStoreEnhanced;

    beforeEach(() => {
      store = mockStore();
    });

    it('fetches app version events', async () => {
      await store.dispatch(fetchAppVersionEvents('app-slug', 'app-version') as any);

      expect(store.getActions()).toMatchSnapshot();
    });

    it("can't fetch app version events", async () => {
      (api.getAppVersionEvents as jest.Mock).mockRejectedValueOnce('api had some issue');
      await store.dispatch(fetchAppVersionEvents('app-slug', 'app-version') as any);

      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
