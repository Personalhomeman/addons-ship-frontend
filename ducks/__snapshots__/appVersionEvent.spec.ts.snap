// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appVersion fetchAppVersionEvents can't fetch app version events 1`] = `
Array [
  Object {
    "type": "TEST_DEVICES_GET_NEXT",
  },
  Object {
    "payload": "api had some issue",
    "type": "TEST_DEVICES_GET_ERROR",
  },
]
`;

exports[`appVersion fetchAppVersionEvents can't fetch test devices 1`] = `
Array [
  Object {
    "type": "TEST_DEVICES_GET_NEXT",
  },
  Object {
    "error": true,
    "payload": [Error: Token not set],
    "type": "TEST_DEVICES_GET_ERROR",
  },
]
`;

exports[`appVersion fetchAppVersionEvents fetches app version events 1`] = `
Array [
  Object {
    "type": "TEST_DEVICES_GET_NEXT",
  },
  Object {
    "type": "TEST_DEVICES_GET_COMPLETE",
  },
]
`;

exports[`appVersion fetchAppVersionEvents fetches test devices 1`] = `
Array [
  Object {
    "type": "TEST_DEVICES_GET_NEXT",
  },
  Object {
    "error": true,
    "payload": [Error: Token not set],
    "type": "TEST_DEVICES_GET_ERROR",
  },
]
`;

exports[`appVersion reducer loads app version events 1`] = `
Array [
  Object {
    "createdAt": 2018-09-20T12:42:31.000Z,
    "id": "1234",
    "logDownloadUrl": "https://www.bitrise.io/assets/svg/logo-bitrise.svg",
    "status": "finished",
    "text": "Event finished successfully!",
    "updatedAt": 2018-09-21T12:42:31.000Z,
  },
  Object {
    "createdAt": 2018-09-22T12:42:31.000Z,
    "id": "1234",
    "logDownloadUrl": "https://www.bitrise.io/assets/svg/logo-bitrise.svg",
    "status": "in-progress",
    "text": "Doing something...",
    "updatedAt": 2018-09-25T12:42:31.000Z,
  },
  Object {
    "createdAt": 2018-09-26T12:42:31.000Z,
    "id": "1234",
    "logDownloadUrl": "https://www.bitrise.io/assets/svg/logo-bitrise.svg",
    "status": "failed",
    "text": "An error occured.",
    "updatedAt": 2018-09-27T12:42:31.000Z,
  },
]
`;

exports[`appVersion reducer loads devices 1`] = `
Array [
  Object {
    "createdAt": 2018-09-20T12:42:31.000Z,
    "id": "1234",
    "logDownloadUrl": "https://www.bitrise.io/assets/svg/logo-bitrise.svg",
    "status": "finished",
    "text": "Event finished successfully!",
    "updatedAt": 2018-09-21T12:42:31.000Z,
  },
  Object {
    "createdAt": 2018-09-22T12:42:31.000Z,
    "id": "1234",
    "logDownloadUrl": "https://www.bitrise.io/assets/svg/logo-bitrise.svg",
    "status": "in-progress",
    "text": "Doing something...",
    "updatedAt": 2018-09-25T12:42:31.000Z,
  },
  Object {
    "createdAt": 2018-09-26T12:42:31.000Z,
    "id": "1234",
    "logDownloadUrl": "https://www.bitrise.io/assets/svg/logo-bitrise.svg",
    "status": "failed",
    "text": "An error occured.",
    "updatedAt": 2018-09-27T12:42:31.000Z,
  },
]
`;
